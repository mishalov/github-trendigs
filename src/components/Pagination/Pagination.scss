.pagination {
  display: flex;
  align-items: center;
  justify-content: center;
}

.pagination-item {
  margin: 8px;
  &__placeholder {
    text-align: center;
    width: 32px;
  }
}
