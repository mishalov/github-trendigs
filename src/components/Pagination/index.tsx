import React from "react";
import Button from "../Button";
import { Link } from "react-router-dom";
import "./Pagination.scss";

interface IPaginationProps {
  pageNow: number;
}

const Pagination: React.FC<IPaginationProps> = ({ pageNow }) => (
  <div className="pagination">
    Page:
    <div className="pagination-item">
      <Link to={"/1"}>
        <Button status={pageNow === 1 || 0 ? "disabled" : "normal"}>1</Button>
      </Link>
    </div>
    <div className="pagination-item">
      {pageNow > 3 ? (
        <div className="pagination-item__placeholder"> ... </div>
      ) : (
        <Link to={"/2"}>
          <Button status={pageNow === 2 ? "disabled" : "normal"}>2</Button>
        </Link>
      )}
    </div>
    <div className="pagination-item">
      {pageNow > 3 ? (
        <Link to={`${pageNow - 1}`}>
          <Button>{pageNow - 1}</Button>
        </Link>
      ) : (
        <Link to={"/3"}>
          <Button status={pageNow === 3 ? "disabled" : "normal"}>3</Button>
        </Link>
      )}
    </div>
    <div className="pagination-item">
      {pageNow > 3 ? (
        <Link to={`${pageNow}`}>
          <Button status="disabled">{pageNow}</Button>
        </Link>
      ) : (
        <Link to={"/4"}>
          <Button status={"normal"}>4</Button>
        </Link>
      )}
    </div>
    <div className="pagination-item">
      {pageNow > 3 ? (
        <Link to={`${pageNow + 1}`}>
          <Button>{pageNow + 1}</Button>
        </Link>
      ) : (
        <Link to={"/5"}>
          <Button status={"normal"}>5</Button>
        </Link>
      )}
    </div>
  </div>
);

export default Pagination;
