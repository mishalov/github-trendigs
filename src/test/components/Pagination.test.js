import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import Pagination from "../../components/Pagination";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

beforeEach(() => {
  wrappers.push(shallow(<Pagination pageNow={0}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={1}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={2}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={3}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={4}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={5}>test</Pagination>));
  wrappers.push(shallow(<Pagination pageNow={200}>test</Pagination>));
});

describe("<Loader /> rendering", () => {
  it("renders correctly", () => {
    expect(wrappers[0]).toMatchSnapshot();
  });
  it("render at firts page correctly", () => {
    expect(wrappers[1]).toMatchSnapshot();
  });
  it("render at page 2 correctly", () => {
    expect(wrappers[2]).toMatchSnapshot();
  });
  it("render at page 3 correctly", () => {
    expect(wrappers[3]).toMatchSnapshot();
  });
  it("render at page 4 correctly", () => {
    expect(wrappers[4]).toMatchSnapshot();
  });
  it("render at page 5 correctly", () => {
    expect(wrappers[5]).toMatchSnapshot();
  });
  it("render at page 200 correctly", () => {
    expect(wrappers[6]).toMatchSnapshot();
  });
});
