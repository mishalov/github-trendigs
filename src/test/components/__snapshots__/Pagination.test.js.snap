// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> rendering render at firts page correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="disabled"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/2"
    >
      <Button
        status="normal"
      >
        2
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/3"
    >
      <Button
        status="normal"
      >
        3
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/4"
    >
      <Button
        status="normal"
      >
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/5"
    >
      <Button
        status="normal"
      >
        5
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering render at page 2 correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/2"
    >
      <Button
        status="disabled"
      >
        2
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/3"
    >
      <Button
        status="normal"
      >
        3
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/4"
    >
      <Button
        status="normal"
      >
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/5"
    >
      <Button
        status="normal"
      >
        5
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering render at page 3 correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/2"
    >
      <Button
        status="normal"
      >
        2
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/3"
    >
      <Button
        status="disabled"
      >
        3
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/4"
    >
      <Button
        status="normal"
      >
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/5"
    >
      <Button
        status="normal"
      >
        5
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering render at page 4 correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <div
      className="pagination-item__placeholder"
    >
       ... 
    </div>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="3"
    >
      <Button>
        3
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="4"
    >
      <Button
        status="disabled"
      >
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="5"
    >
      <Button>
        5
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering render at page 5 correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <div
      className="pagination-item__placeholder"
    >
       ... 
    </div>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="4"
    >
      <Button>
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="5"
    >
      <Button
        status="disabled"
      >
        5
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="6"
    >
      <Button>
        6
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering render at page 200 correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <div
      className="pagination-item__placeholder"
    >
       ... 
    </div>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="199"
    >
      <Button>
        199
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="200"
    >
      <Button
        status="disabled"
      >
        200
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="201"
    >
      <Button>
        201
      </Button>
    </Link>
  </div>
</div>
`;

exports[`<Loader /> rendering renders correctly 1`] = `
<div
  className="pagination"
>
  Page:
  <div
    className="pagination-item"
  >
    <Link
      to="/1"
    >
      <Button
        status="normal"
      >
        1
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/2"
    >
      <Button
        status="normal"
      >
        2
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/3"
    >
      <Button
        status="normal"
      >
        3
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/4"
    >
      <Button
        status="normal"
      >
        4
      </Button>
    </Link>
  </div>
  <div
    className="pagination-item"
  >
    <Link
      to="/5"
    >
      <Button
        status="normal"
      >
        5
      </Button>
    </Link>
  </div>
</div>
`;
